font-size declarations don’t support text resizing (WCAG 2.0 SC 1.4.4)

Testing Tacit on the [online demo](https://yegor256.github.io/tacit/), I noticed font-size declarations are set in `px`. This is problematic for people who rely on browsers’ capability to enlarge text. As-is, Tacit is unusable on sites that must comply with the accessibility standard WCAG 2.0 AA (or WCAG 2.1 AA), as a failure of [SC 1.4.4 Resize Text](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html).

Here is a quick demo. Note how the contents of example.com resize as I change my browser settings, while the Tacit page doesn’t change:

![Side by side screencast of example.com, the Tacit demo, and Chrome settings. Changing the font size in Chrome settings only changes example.com’s font size](https://user-images.githubusercontent.com/877585/109787786-7fc1fe00-7c06-11eb-875e-3d018ddac288.gif)

---

All of Tacit’s font-size declarations should be in `rem` (or `em`), so the text reflects the font-size changes done by the browser on the `html` element.

I think the simplest way to do this could be to change the `$em` declaration to be in `rem`: https://github.com/yegor256/tacit/blob/23c6ffb037d4942ae20e444f8ae0aad66f1ce037/scss/_defs.scss#L24

```scss
$em: 1.125rem; // 18 px.
```

$em is also used for vertical and horizontal spacing, which I’m not sure we’d always want to resize as well, so it’d also be worth considering whether there should be two separate "size" units, one in `rem` for things that should enlarge with text, and one in `px` for things that should keep their size.

There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@9614737e-407d-430f-8585-2bd187b84f39, thread: pool-364-thread-1
org.hamcrest.MatcherAssert[20] java.lang.AssertionError: HTTP response status is not equal to 201:
404 Not Found [https://api.github.com/repos/yegor256/tacit/labels]
Server: GitHub.com
Date: Wed, 03 Mar 2021 09:59:29 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
X-OAuth-Scopes: admin:enterprise, admin:gpg_key, admin:org, admin:org_hook, admin:public_key, admin:repo_hook, delete:packages, delete_repo, gist, notifications, repo, user, workflow, write:discussion, write:packages
X-Accepted-OAuth-Scopes: 
X-GitHub-Media-Type: github.v3
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4558
X-RateLimit-Reset: 1614767837
X-Ratelimit-Used: 442
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
Vary: Accept-Encoding, Accept, X-Requested-With
X-GitHub-Request-Id: CF28:5FE8:559EFA:DB0F07:603F5E01

{"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/issues#create-a-label"}
Expected: HTTP response with status 201
     but: was <404 Not Found [https://api.github.com/repos/yegor256/tacit/labels]
Server: GitHub.com
Date: Wed, 03 Mar 2021 09:59:29 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
X-OAuth-Scopes: admin:enterprise, admin:gpg_key, admin:org, admin:org_hook, admin:public_key, admin:repo_hook, delete:packages, delete_repo, gist, notifications, repo, user, workflow, write:discussion, write:packages
X-Accepted-OAuth-Scopes: 
X-GitHub-Media-Type: github.v3
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4558
X-RateLimit-Reset: 1614767837
X-Ratelimit-Used: 442
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
Vary: Accept-Encoding, Accept, X-Requested-With
X-GitHub-Request-Id: CF28:5FE8:559EFA:DB0F07:603F5E01

{"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/issues#create-a-label"}>
```

[0.55.1](https://github.com/zerocracy/farm): Issue: yegor256/tacit#216, Action: `opened`

--- a/scss/_defs.scss
+++ b/scss/_defs.scss
@@ -22,6 +22,7 @@
 
 // sizes
 $em: (18 / 16) * 1rem;
+$spacing: 18px;
 
 // fonts
 $sans-serif: system-ui, 'Helvetica Neue', Helvetica, Arial, sans-serif;

--- a/scss/_defs.scss
+++ b/scss/_defs.scss
@@ -21,7 +21,7 @@
 // SOFTWARE.
 
 // sizes
-$em: 18px;
+$em: (18 / 16) * 1rem;
 
 // fonts
 $sans-serif: system-ui, 'Helvetica Neue', Helvetica, Arial, sans-serif;

--- a/scss/_forms.scss
+++ b/scss/_forms.scss
@@ -38,7 +38,7 @@ textarea,
 select,
 button {
   @extend %sans-serif;
-  border-radius: .2 * $em;
+  border-radius: .2 * $spacing;
   display: inline-block;
   padding: .55 * $em;
 
@@ -147,7 +147,7 @@ input[type="radio"] {
   flex-grow: 0;
   height: 1.65 * $em;
   margin-left: 0;
-  margin-right: .5 * $em;
+  margin-right: .5 * $spacing;
   vertical-align: middle;
 
   + label {
@@ -156,5 +156,5 @@ input[type="radio"] {
 }
 
 select[multiple] {
-  min-width: 15 * $em;
+  min-width: 15 * $spacing;
 }

--- a/scss/_grid.scss
+++ b/scss/_grid.scss
@@ -33,7 +33,6 @@
 
 html,
 body {
-  @extend %sans-serif;
   width: 100%;
 }
 
@@ -42,6 +41,7 @@ html {
 }
 
 body {
+  @extend %sans-serif;
   background: $white;
   color: lighten($black, 10);
   padding: 2 * $em;

--- a/scss/_grid.scss
+++ b/scss/_grid.scss
@@ -44,7 +44,7 @@ body {
   @extend %sans-serif;
   background: $white;
   color: lighten($black, 10);
-  padding: 2 * $em;
+  padding: 2 * $spacing;
 }
 
 p,
@@ -65,7 +65,7 @@ address {
 section {
   margin-left: auto;
   margin-right: auto;
-  width: 50 * $em;
+  width: 50 * $spacing;
 }
 
 aside {
@@ -76,7 +76,7 @@ aside {
 article,
 header,
 footer {
-  padding: 2.4 * $em;
+  padding: 2.4 * $spacing;
 }
 
 article {
@@ -94,8 +94,8 @@ nav {
 
     li {
       display: inline-block;
-      margin-left: .5 * $em;
-      margin-right: .5 * $em;
+      margin-left: .5 * $spacing;
+      margin-right: .5 * $spacing;
       vertical-align: middle;
 
       &:first-child {
@@ -111,7 +111,7 @@ nav {
 
 ol,
 ul {
-  margin-left: 1.75 * $em;
+  margin-left: 1.75 * $spacing;
 }
 
 li {

--- a/scss/_pre.scss
+++ b/scss/_pre.scss
@@ -36,7 +36,7 @@ pre {
   border-left: .1 * $em solid lighten($green, 25);
   line-height: 1.4 * $em;
   overflow: auto;
-  padding-left: $em;
+  padding-left: $spacing;
 
   code {
     background: none;
@@ -49,7 +49,7 @@ pre {
 code,
 kbd {
   background: lighten($green, 60);
-  border-radius: .2 * $em;
+  border-radius: .2 * $spacing;
   color: $green;
   display: inline-block;
   line-height: $em;

--- a/scss/_responsive.scss
+++ b/scss/_responsive.scss
@@ -22,17 +22,17 @@
 
 @media (max-width: $mobile-width) {
   body {
-    padding: $em 0;
+    padding: $spacing 0;
   }
 
   article {
     border: 0;
-    padding: $em;
+    padding: $spacing;
   }
 
   header,
   footer {
-    padding: $em;
+    padding: $spacing;
   }
 
   textarea,

